### PR TITLE
Add python tests for contents handler

### LINKED ISSUE #1542: Add python tests for contents handler
**Describe the issue**
During the review of #1460 we realized the existing python tests don't exercise the contents handler. Since the handler is exercised via the integration tests and a release is pending, we decided to hold off including those tests (which can be a bit more involved) in #1460 and performed some additional manual tests using `curl` etc. in the meantime.

This issue exists to ensure the handler tests get implemented.  These tests will likely follow the model set in the metadata handler tests.

### PR DESCRIPTION
**Pull request by kiersten-stokes**
Fixes #1542. Currently tests the following scenarios:

- [x] File not found
- [x] File is a directory
- [x] Invalid POST request
- [x] Invalid file extension
- [x] Notebook file (with python kernel) with env vars
- [x] Python file with env vars
- [x] R file with env vars

Additional scenarios that we may want to test:

- [x] Notebook with no env vars
- ~~Python file with no env vars~~ (general case covered by Notebook 'no-env' test)
- ~~R file with no env vars~~ (general case covered by Notebook 'no-env' test)
- ~~Notebooks with an R kernel~~ (covered by unit tests)

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: elyra/contents/tests/test_handlers.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    filepath = "nofile.py"
+
+    with pytest.raises(HTTPClientError) as e:
+        await jp_fetch('elyra', 'contents/properties', filepath)
+    assert expected_http_error(e, 404)
+
+
+async def test_file_is_not_directory(jp_fetch, create_directory, directory_name):
+    with pytest.raises(HTTPClientError) as e:
+        await jp_fetch('elyra', 'contents/properties', 'dir.py')
```
Comment (kevin-bates) [inline elyra/contents/tests/test_handlers.py:33]: Should the reference to `'dir.py'` be replaced with `directory_name`?

File: elyra/contents/tests/test_handlers.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    assert expected_http_error(e, 400)
+
+
+async def test_invalid_post_request(jp_fetch):
+    response = await jp_fetch('elyra', 'contents/properties', 'dir.py', body=json.dumps(""), method='POST')
```
Comment (kevin-bates) [inline elyra/contents/tests/test_handlers.py:38]: Same: `'dir.py'` -> `directory_name`?  Will need to add the fixture name to the test method.

File: elyra/contents/tests/test_utils.py [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+# limitations under the License.
+#
+import json
+import tornado
+import os
+import errno
+
+
+def expected_http_error(error, expected_code, expected_message=None):
+    """Check that the error matches the expected output error."""
```
Comment (kevin-bates) [inline elyra/contents/tests/test_utils.py:23]: As of #1551 we import this function `from jupyter_server.tests.utils import expected_http_error`.

**Review by kevin-bates — COMMENTED** [review]
This is looking good, just had a couple of initial comments.

File: elyra/contents/tests/test_handlers.py (hunk 1 above)
Comment (kiersten-stokes) [reply elyra/contents/tests/test_handlers.py:33]: Ah yes, thank you for catching that! I clearly changed the implementation halfway through

File: elyra/contents/tests/test_utils.py (hunk 3 above)
Comment (kiersten-stokes) [reply elyra/contents/tests/test_utils.py:23]: A very fresh change! Will get that fixed

Comment (kevin-bates) [reply elyra/contents/tests/test_utils.py:23]: Yes, about as fresh as they come.

**Comment by kevin-bates** [comment]
@kiersten-stokes - The latest changes look good.  Were you going to add the additional tests?  I think, at a minimum, having at least one test that is a valid file w/o envs should be added since the others are tested via unit tests.

**Comment by kiersten-stokes** [comment]
@kevin-bates Yeah for those additional tests I'm basically looking for input/opinions on whether they should be added. I'm happy to add them all if you think it would be best

**Comment by kevin-bates** [comment]
I don't think they're all necessary, just the general "topic" of each seems sufficient.  In this case, what's missing is a valid file (of whatever type) with no envs to be parsed.

**Comment by kiersten-stokes** [comment]
Makes sense! I'll do one 'no envs' example. Any thoughts on an R-based notebook test? I feel like it's probably not necessary since the non-handler tests cover that scenario

**Comment by kevin-bates** [comment]
Yes, I agree - the direct unit tests cover that just fine.

**Review by kevin-bates — APPROVED** [review]
Looks good - thanks Kiersten.
